sdk/storage/azure-storage-file-datalake/samples/datalake_samples_access_control_recursive_async.py

```diff
@@ -58,9 +60,19 @@ async def progress_callback(acl_changes):
         failed_entries.append(acl_changes.batch_failures)
 
     # illustrate the operation by using a small batch_size
-    acl_change_result = await directory_client.set_access_control_recursive(acl=acl,
-                                                                            progress_callback=progress_callback,
-                                                                            batch_size=5)
+    try:
+        acl_change_result = await directory_client.set_access_control_recursive(acl=acl,
+                                                                                progress_callback=progress_callback,
+                                                                                batch_size=5)
+    except AzureError:
```

Comment:
It might be best to demonstrate this pattern in a loop of some kind? In the case that this long-running operation fails multiple times over a 24 period, we would want to keep catching and continuing.

Reply:
In the case that this long-running operation fails multiple times, the retry policy would handle this.
The sample is just demonstrating after retry there was still an exception thrown, how they can use the continuation_token attr of AzureError to restart the operation. So we might not need to use a while loop, what do you think?